### PR TITLE
New blog post: Chaos Testing Arc SQL MI High Availability

### PR DESCRIPTION
**Pull request by mdrakiburrahman**
`index.mdx` contains the markdown content.

Accompanied by a small demo:
[IMAGE ALT TEXT HERE](https://www.youtube.com/embed/4OfM7e6fi08)

### CODE REVIEW

File: content/arc-sqlmi-chaos-mesh/index.mdx [hunk 1]
```diff
@@ -0,0 +1,248 @@
...
+
+Please, do not try any of this in Production.
+
+Also, none of the content presented here is official guidance or recommendation from Microsoft. It's for educational (or maybe, entertainment 😄) purposes only.
+
+</Callout>
+
+Synchronous Transaction Replication and resilient replica recovery is something only a handful of Database Products have built up as a core-feature over the years, many top rankers in [this list](https://db-engines.com/en/ranking) do not have an answer to this difficult problem. Even the one's that do have a documented solution, it's pretty hard to compare and contrast how good the implementation _actually_ is from studying docs/watching a demo or conducting a POC, without either:
```
Comment (sfeitler) [inline content/arc-sqlmi-chaos-mesh/index.mdx:26]: Instead of "core-feature over the years, many top rankers" try
"core-feature over the years. Many top rankers"

"one's" should be "ones", no apostrophe, in "Even the one's that do"

File: content/arc-sqlmi-chaos-mesh/index.mdx [hunk 2]
```diff
@@ -0,0 +1,248 @@
...
+
+"_Do whatever you want to the system (besides deleting `mdf/ldf` files etc., AKA no irrecoverable damage), for any duration of time, with the end goal of leaving the system in a state where it cannot self-heal, and requires manual intervention (e.g. intelligently crafted T-SQL commands that must be manually run by some human being)._
+
+_If it requires manual intervention, the system is broken, and that's a major bug._"
+
+</Callout>
+
+This time, I had Kubernetes on my side, with the SQL engine running in disposable Linux containers ([Azure Arc-enabled SQL Managed Instance, Business-Critical Tier](https://learn.microsoft.com/en-us/azure/azure-arc/data/service-tiers#service-tier-comparison)), and the latest-and-greatest Chaos Engineering software from CNCF - called [Chaos Mesh](https://www.cncf.io/projects/chaosmesh/) - offering many novel [Custom Resource Defintions](https://github.com/chaos-mesh/chaos-mesh#chaos-operator) that can be used to declare layers upon layers of controlled, systematic degradations (rather than one-off imperative scripts with a single entry and exit point like, restarting a VM, then moving on to cause some other mischief in a linear fashion, etc.).
```
Comment (sfeitler) [inline content/arc-sqlmi-chaos-mesh/index.mdx:43]: "point like, restarting a VM, then moving"

I would remove at least the first comma. The second is optional.

File: content/arc-sqlmi-chaos-mesh/index.mdx [hunk 3]
```diff
@@ -0,0 +1,248 @@
...
+
+Having [pondered about questions like this](https://www.reddit.com/r/kubernetes/comments/tfga4y/eli5_why_is_it_not_advisable_to_run_databases_in/) about the scary unknowns of hosting data inside ephemeral containers (K8s Pods), I was feeling pretty good about my chances of discovering a failure-mode that triumphed `sqlservr.exe`(s).
+
+## High Availability Architecture and possible faliure points
+
+<Callout>
+
+The key to try and make a system miserably fail is to first understand it's architecture and key lifelines. That way, the kinds of failures we can produce _really_ try to hurt the system by cutting off one or more of those lifelines.
```
Comment (sfeitler) [inline content/arc-sqlmi-chaos-mesh/index.mdx:51]: "first understand it's architecture"

No apostrophe in this version of "its".

File: content/arc-sqlmi-chaos-mesh/index.mdx [hunk 4]
```diff
@@ -0,0 +1,248 @@
...
+
+</Callout>
+
+### HA Kubernetes Operator and Pendulum Analogy
+
+- Azure Arc-enabled SQL Managed Instance Business Critical tier, besides wrapping all SQL replicas behind an Availability Group Listener, comes with a dedicated [Kubernetes Operator](https://kubernetes.io/docs/concepts/extend-kubernetes/operator/) called the `arc-ha-orchestrator`.
+
+  - An Operator is a dedicated piece of computer program responsible for monitoring the health state of some system it "operates", based on some system specific [Finite-state machine](https://en.wikipedia.org/wiki/Finite-state_machine) definition (basically, business logic).
+
+    It then uses the observed signals to figure out the Current-State, and perform some small, pre-defined, intelligent actions to "nudge" (rather than force), the system towards a healthy (i.e. Desired) state. That system could be anything, SQL Server, a pendulum, a self-driving car etc.
```
Comment (sfeitler) [inline content/arc-sqlmi-chaos-mesh/index.mdx:61]: ""nudge" (rather than force), the system"
You shouldn't have a comma here

Comment (sfeitler) [reply content/arc-sqlmi-chaos-mesh/index.mdx:61]: "anything, SQL Server" change that comma to a colon to start a list:

"anything: SQL Server"

File: content/arc-sqlmi-chaos-mesh/index.mdx [hunk 5]
```diff
@@ -0,0 +1,248 @@
...
+
+    - Kubernetes Operators are a very elaborate application of [PID Controllers in Electronics](https://en.wikipedia.org/wiki/PID_controller#Fundamental_operation) that have been around since 1920s. Actually, all of Kubernetes is built on applying this idea as code.
+
+  - The `arc-ha-orchestrator`'s sole responsibility is doing everything within it's power to monitor and maintain the health of the SQL Server Availability Group it manages. It is the operator's life's mission to ensure the AG never gets into an irrecoverable, i.e. corrupt state. If it senses this is about to happen, it will do whatever actions it has within it's arsenal, to ensure the State Machine never gets into a corrupt state that cannot be recovered from. The way this is implemented is by applying the [Paxos protocol](<https://en.wikipedia.org/wiki/Paxos_(computer_science)>).
```
Comment (sfeitler) [inline content/arc-sqlmi-chaos-mesh/index.mdx:65]: "within it's power"

This should be

"within its power"

Comment (sfeitler) [reply content/arc-sqlmi-chaos-mesh/index.mdx:65]: Also "operator's life's mission" can just be "operator's life mission"

File: content/arc-sqlmi-chaos-mesh/index.mdx (hunk 1 above)
Comment (mdrakiburrahman) [reply content/arc-sqlmi-chaos-mesh/index.mdx:26]: Got it!

File: content/arc-sqlmi-chaos-mesh/index.mdx (hunk 2 above)
Comment (mdrakiburrahman) [reply content/arc-sqlmi-chaos-mesh/index.mdx:43]: Done

File: content/arc-sqlmi-chaos-mesh/index.mdx (hunk 4 above)
Comment (mdrakiburrahman) [reply content/arc-sqlmi-chaos-mesh/index.mdx:61]: Done and done

File: content/arc-sqlmi-chaos-mesh/index.mdx (hunk 5 above)
Comment (mdrakiburrahman) [reply content/arc-sqlmi-chaos-mesh/index.mdx:65]: Done *3
